### PR TITLE
update vendor to latest 1.4 origin release

### PR DESCRIPTION
**Pull request by elmiko**
This change updates the vendor directory to use the latest OpenShift
Origin sources for the 1.4 release line. It also adds the Godeps.json
file to the top level Godeps directory. This change has been added to
help improve the quality of the dependency chain for the oshinko cli and
rest applications.

In addition to the vendor changes, the Godeps.json file has been
modified by hand to include the dependencies for the rest server. These
changes have been made by hand to account for difficulties with the
godep tooling when evaluating the OpenShift code base. As OpenShift
Origin uses modified upstreams in its vendor directory, these have been
included in the oshinko vendor.

The CONTRIBUTING.md file has also been updated to record the process
used to create the vendor directory and the Godeps.json file.

### CODE REVIEW

**Comment by elmiko** [comment]
i think we will need to plan out the upgrade to the 1.5 origin sources as it appears some of the kubernetes client stuff we are depending on has moved in the upstream. i propose this as a starting point to get our dependencies tracked, next we can pivot to address the new stuff.

i think it's worth noting that the oshinko-cli project will build and run solely from the openshift vendor folder. it is only the rest server that requires anything extra at this point. i think we should reconsider if keeping the rest server in place is worth it, or if it would be easier to maintain as a separate repo.

**Comment by elmiko** [comment]
@tmckayus @sub-mod ptal

File: rest/CONTRIBUTING.md [hunk 1]
```diff
@@ -46,6 +46,32 @@ In this manner, we can ensure that the dependency chain for the oshinko-rest
...
+manually by copying them from github into the appropriate locations in the
+vendor directory and their entries added to the `Godeps.json` file.
+
+*Dependency creation steps*
+1. copy `vendor` directory from `openshift/origin` to root of oshinko-cli
+1. copy `Godeps` directory from `openshift/origin` to root of oshinko-cli
+1. copy `openshift/origin` to `./vendor/github.com/openshift/`
+1. remove `vendor`, `Godeps` and `.git` directories from the copied
+   `./vendor/github.com/openshift/origin/` directory
+1. add rest dependencies to `vendor` directory and `Godeps.json` file
```
Comment (tmckayus) [inline rest/CONTRIBUTING.md]: should we list what the extra REST dependencies are at the moment, or at least how to determine what they are?

Comment (elmiko) [inline rest/CONTRIBUTING.md]: great question, i had considered making a separate document in the rest directory for those godeps. i will add something

**Comment by elmiko** [comment]
updated to add some text mentioning the rest dependencies in the CONTRIBUTING.md doc

**Comment by tmckayus** [comment]
lgtm

**Pull request merged**